### ISSUE TITLE
Posting individual items issue with large amount

### ISSUE THREAD

**Issue by jlmk99** [shirsig/aux-addon#330]
When posting individual items that you have a lot of sometimes causes internal errors. It stops the posting leaving you with either the wrong value or the right value left in your bags. Once it errors out you can continue reposting as long as it matches the total in your bag. If it doesn't aux will lock out posting and require a /reload. So to avoid that you continuously have to make the amount match in aux to what's left in your bag. Sometimes when reposting it either errors out within a few posts or again later. It's quite frustrating tbh. I mostly post green quality gems individually that I have in large quantity. Usually on smaller quantities it's not a problem. Like I said though once the error occurs you can post say 3 out of 40 it'll error out.

**Comment by Witnessthis** [comment]
I'm also experiencing this issue.

**Comment by mattk926** [comment]
Same issue here! Has been this way for months!

**Comment by shirsig** [comment]
you mean posting many times the same items with small stack size? Since tbc this is actually not done by aux anymore (wouldn't be possible, you would have to click for each individual posting) but through a single api call so I don't think there's anything I can do about that.

**Comment by mattk926** [comment]
Example: I have 500 elixirs. I want to sell them in stacks of 5. This would be 100 listings of 5 I click post and then it starts posting them. depending how many queries were made in the past x seconds to the AH API blizzard starts throttling the postings. once that happens that posting stops and errors out. The addon may or may not have kept track of how many items are still in inventory and you may have to close/reopen for aux to re"scan" the inventory to determine how many elixirs are actually in your inventory (minor inconvenience). If there was a way to "slow down" the posting process of each individual stack it might help alleviate the throttling interruptions experienced while trying to post a large amount of items in a short amount of time.

**Comment by mattk926** [comment]
This also happens more frequently directly after the posting scan happens on an item with a lot of listings.

**Comment by shirsig** [comment]
> 
> 
> Example: I have 500 elixirs. I want to sell them in stacks of 5. This would be 100 listings of 5 I click post and then it starts posting them.

This actually is done using a single call to the blizzard API: PostAuction(start_price, buyout_price, duration, stack_size, stack_count)
This is a function which can only be called with a hardware event so there is no way to throttle it while keeping it fully automatic, and if it's not gonna be automatic, well, you can already do that if you just reduce the stack count and post multiple times.

**Comment by Witnessthis** [comment]
Is there perhaps a built in throttle in the blizzard api to reduce AH lag? For example being allowed to post X items per Y time. And the symptoms of this is what we see?

**Comment by mattk926** [comment]
Yes this is what we are running into. Blizzard has a limit set on the AH to help reduce the "lag" experienced. just not sure what the appropriate solution is. Part of the nuisance is the addon gets out of sync when this throttle happens forcing close and open of the addon again to have it pull information from the inventory to be accurate. The other part is the unknown number or limit. so it can be annoying to post something right after a "post scan" which if you try to post items during the scan if it is an item which has a large amount of auctions it will error out right away.

**Comment by Witnessthis** [comment]
I guess it would be possible to detect if a throttle happend based on how many items you expected to post and how many you still have remaining. (If no other prettier solution exist). Forcing a recount of the items is probably necessary at this point to not have to restart the addon. I've also experienced situations in which the UI bugs such that it's impossible to press the post button since it's greyed out even after a close/open. To fix this issue I had to reload the ui.
